edgelet/iotedge/src/support_bundle.rs

```diff
@@ -75,45 +87,80 @@ where
 {
     pub fn new(
         log_options: LogOptions,
-        location: OsString,
         include_ms_only: bool,
         verbose: bool,
         iothub_hostname: Option<String>,
+        output_location: OutputLocation,
         runtime: M,
     ) -> Self {
         SupportBundle {
             runtime,
             log_options,
-            location,
             include_ms_only,
             verbose,
             iothub_hostname,
+            output_location,
         }
     }
 
-    fn make_state(self) -> Result<BundleState<M>, Error> {
+    fn make_file_state(self) -> Result<BundleState<M, File>, Error> {
         let file_options =
             zip::write::FileOptions::default().compression_method(zip::CompressionMethod::Deflated);
-        let location = PathBuf::from(&self.location);
 
         let zip_writer = zip::ZipWriter::new(
-            File::create(&location)
+            File::create(Path::new(self.output_location.get_file_location()))
                 .map_err(|err| Error::from(err.context(ErrorKind::SupportBundle)))?,
         );
 
         Ok(BundleState {
             runtime: self.runtime,
             log_options: self.log_options,
-            location,
             include_ms_only: self.include_ms_only,
             verbose: self.verbose,
             iothub_hostname: self.iothub_hostname,
             file_options,
             zip_writer,
+            output_location: self.output_location,
+        })
+    }
+
+    fn make_vector_state(self) -> Result<BundleState<M, Cursor<Vec<u8>>>, Error> {
+        let file_options =
+            zip::write::FileOptions::default().compression_method(zip::CompressionMethod::Deflated);
+
+        let zip_writer = zip::ZipWriter::new(Cursor::new(Vec::new()));
```

Comment:
Should we worry that this project is no longer actively maintained? https://github.com/mvdnes/zip-rs

Reply:
This means that we must track security issues/bugs/etc, right?

Reply:
This could be a possible alternative, it has a very similar interface: https://crates.io/crates/mini-fs. However, it has 900 lifetime downloads vs 1.1 million: https://crates.io/crates/zip. I think it makes sense to stick with zip for now.

Reply:
Perhaps, but I am still concerned with the support burdeon

Reply:
Talked to arnav, he said it's ok for now